storage/s3/src/main/java/io/aiven/kafka/tieredstorage/storage/s3/S3UploadOutputStream.java

```diff
@@ -185,14 +227,18 @@ private void abortUpload() {
             .uploadId(uploadId)
             .build();
         client.abortMultipartUpload(request);
-        closed = true;
     }
 
     private void flushBuffer(final ByteBuffer buffer,
-                             final int actualPartSize) {
-        try (final InputStream in = new ByteBufferInputStream(buffer)) {
+                             final int actualPartSize,
+                             final boolean multiPartUpload) {
+        try (final InputStream in = new ByteBufferMarkableInputStream(buffer)) {
```

Comment:
@jiafu1115 could you elaborate on the need for a markable inputstream here? Why is `ByteBufferInputStream` not enough?

Reply:
ping @jiafu1115 if you could have a look at this one.